internal/api/genapi/templates.go

```diff
@@ -299,16 +305,40 @@ func (c *Client) List(ctx context.Context, {{ .CollectionFunctionArg }} string,
 		if apiErr != nil {
 			return nil, apiErr
 		}
-		target.Items = append(target.Items, page.Items...)
+		for _, item := range page.Items {
+			if i, ok := idToIndex[item.Id]; ok {
+				// Item has already been seen at index i, update in-place
+				target.Items[i] = item
+			} else {
+				target.Items = append(target.Items, item)
+				idToIndex[item.Id] = len(target.Items)-1
+			}
+		}
 		target.RemovedIds = append(target.RemovedIds, page.RemovedIds...)
 		target.EstItemCount = page.EstItemCount
 		target.RefreshToken = page.RefreshToken
 		target.ResponseType = page.ResponseType
 		target.response = resp
 		if target.ResponseType == "complete" {
```

Comment:
Even in the case where the server returns something strange, the user can cancel the operation via the context. If we inverted this check, it would make it very difficult to introduce another value to the response_type enum (not that we're likely to). I think we can keep this for now and see if we need to change it down the line?